discord/bot.py

```diff
@@ -270,81 +270,172 @@ async def get_desynced_commands(
         """
 
         # We can suggest the user to upsert, edit, delete, or bulk upsert the commands
+        class DefaultComparison:
+            """
+            Comparison rule for when there are multiple default values that should be considered equivalent when comparing 2 objects.
+            Allows for a custom check to be passed for further control over equality.
+
+            Attributes
+            ----------
+            defaults: :class:`tuple`
+                The values that should be considered equivalent to each other
+            callback: Callable[[Any, Any], bool]
+                A callable that will do additional comparison on the objects if neither are a default value.
+                Defaults to a `!=` comparison.
+                It should accept the 2 objects as arguments and return True if they should be considered equivalent
+                and False otherwise.
+            """
+
+            def __init__(
+                self,
+                defaults: tuple[Any, ...],
+                callback: Callable[[Any, Any], bool] = lambda x, y: x != y,
+            ):
+                self.defaults = defaults
+                self.callback = callback
+
+            def _check_defaults(self, local, remote) -> bool | None:
+                defaults = (local in self.defaults) + (remote in self.defaults)
+                if defaults == 2:
+                    # Both are defaults, so they can be counted as the same
+                    return False
+                elif defaults == 0:
+                    # Neither are defaults so the callback has to be used
+                    return None
+                else:
+                    # Only one is a default, so the command must be out of sync
+                    return True
+
+            def check(self, local, remote) -> bool:
+                if (rtn := self._check_defaults(local, remote)) is not None:
+                    return rtn
+                else:
+                    return self.callback(local, remote)
+
+        class DefaultSetComparison(DefaultComparison):
+            def check(self, local, remote) -> bool:
+                try:
+                    local = set(local)
+                except TypeError:
+                    pass
+                try:
+                    remote = set(remote)
+                except TypeError:
+                    pass
+                return super().check(local, remote)
+
+        type NestedComparison = dict[str, NestedComparison | DefaultComparison]
```

Comment:
```suggestion
        NestedComparison: TypeAlias = dict[str, NestedComparison | DefaultComparison]
```
The `type` keyword is only supported since py 3.12